### ISSUE TITLE
Accept dragged images from browser

### ISSUE THREAD

**Issue by kriomant** [fabiospampinato/notable#15]
There is site with image which is link. I drag image to Notable, but it inserts link, not image. EverNote prefers image in such situation, and I used to it. Please either allow to select what to insert (with keyboard modifiers / popup menu / etc) or make some priorities.

It would be also cool if notes list accepted dragging onto it too. It should just create new note with dragged item as the only content.

Platform: OSX

**Comment by fabiospampinato** [comment]
Yes drag & drop should be better supported. IMHO we should support the following:

- [ ] Dragging a file/image over the editor adds it as an attachment
- [ ] Dragging a note over a tag adds that tag to the note
- [ ] Dragging a supported file into the main bar imports that note

**Comment by andreferreirav2** [comment]
On the subject of attachment insertion, it would be great to insert the whole line `[](@attachment/<attachment_name>.png)` for already existing ones, simply by dragging it from the *clip icon dropdown* to the text.

<img width="178" alt="screenshot 2018-12-31 10 14 12" src="https://user-images.githubusercontent.com/2427597/50558305-dceb5d80-0ce4-11e9-8638-f8dbc7384c38.png">

**Comment by andreferreirav2** [comment]
Another cool improvement on the Drag&Drop would be to stop duplicates from being uploaded, ex: drag&drop the same file twice for the sake of convenience, but the editor only uploads it once to the attachments and reuses it the 2nd time.

This could be achieved via md5 hash of the files or similar, since a copy of the file can be dragged from different locations but is still the same file.

**Comment by fabiospampinato** [comment]
> On the subject of attachment insertion, it would be great to insert the whole line [](@attachment/<attachment_name>.png) for already existing ones, simply by dragging it from the clip icon dropdown to the text.

This may be tricky to implement, but it sounds like a good idea.

> Another cool improvement on the Drag&Drop would be to stop duplicates from being uploaded

I agree that you probably never need 2 copies of the same file in the same note, but I'm not sure how many people will take advantage of this functionality, if it were to be implemented 🤔For the time being I'm focusing on some other more useful features.

**Comment by abijr** [comment]
Can we add `Drag and drop note into editor creates a note link` into the support checklist?

**Comment by fabiospampinato** [comment]
@abijr thanks for the suggestion, I've added `Dragging a tag/note/attachment into the editor should embed it` which is a bit more general.

**Comment by aravindhsampath** [comment]
+1 for drag & drop of images into notes. 

I have been using Apple Notes for all my note taking. A key thing I do while taking notes is add screenshots and add clips of diagrams from websites. I have never found an alternative to Notes that gave me that ease of drag & drop images. If notable could do that I would switch to this entirely. 

Thanks and cheers for creating this project!

**Comment by fabiospampinato** [comment]
I just added a few more to the list:

- [ ] Dragging a data directory into the app's icon opens that data directory in a new window
- [ ] Dragging a note out of the app should export it
- [ ] Dragging an attachment out of the app should export it

**Comment by xplatinum** [comment]
I would love to be able to select multiple notes and drag and drop them to a tag/notebook.

**Comment by fabiospampinato** [comment]
> I would love to be able to select multiple notes and drag and drop them to a tag/notebook.

I've just added a mention to that to my original post above 👍

**Comment by NiccoloGranieri** [comment]
Been following the invisible thread between issues and arrived here! I was wondering if, in the timeline that is considering to add autocomplete to Notebooks and tags, there is the idea of implementing the default _press tab to autocomplete_ feature of the terminal window in macOS and many other editors.

**Comment by fabiospampinato** [comment]
@NiccoloGranieri we are tracking that mostly here: #20

**Comment by vict0rsch** [comment]
Hey, thanks for the great work :) 

I've been following threads for a while now, do you have any kind of estimation of when these drag/drop files/tags etc. issues will come to life @fabiospampinato ?

Thanks!

**Comment by fabiospampinato** [comment]
@vict0rsch My best eta estimate would be about 4 months, but take that with a bag of salt.
